dapp_runner/runner/service.py

```diff
@@ -79,26 +81,43 @@ async def _wait_for_termination(self):
             self._report_state_change()
 
 
+class HttpProxyDappService(DappService, HttpProxyService):
+    """Yapapi Service definition enabling HTTP proxy for dapps."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+
 async def get_service(
     name: str, desc: ServiceDescriptor, payloads: Dict[str, Payload]
-) -> Type[DappService]:
+) -> Tuple[Type[Service], dict]:
     """Create a service class corresponding with its descriptor."""
 
     try:
         payload_instance = payloads[desc.payload]
     except KeyError:
         raise RunnerError(f"Undefined payload: `{desc.payload}`")
 
-    async def get_payload():
-        return payload_instance
+    service_instance_params: dict = {}
+    service_instance_params["entrypoint"] = desc.entrypoint
 
     DappServiceClass = type(
         f"DappService-{name}",
         (DappService,),
-        {
-            "get_payload": staticmethod(get_payload),
-            "entrypoint": desc.entrypoint,
-        },
+        {},
     )
 
-    return DappServiceClass
+    if desc.http_proxy:
+        port_mapping = desc.http_proxy.ports[0]
+        service_instance_params["remote_port"] = port_mapping.remote_port
+        DappServiceClass = type(
+            f"DappService-{name}",
+            (HttpProxyDappService,),
+            {},
+        )
+
+    run_service_kwargs: dict = {}
```

Comment:
well, actually, it surprised me that we're generating the whole dictionary of args for `run_service` here... but maybe that's better... I somehow expected the dictionary to not be passed as-is into `run_service` and expected some post-processing that would put the payload and instance_kwargs as parameters for `run_service` ... but maybe your solution is better... just thinking whether it's going to be futureproof...